### ISSUE TITLE
QuarkusSmallRyeTracingDynamicFeature bean is removed during build while needed

### ISSUE THREAD

**Issue by sclorng** [quarkiverse/quarkus-operator-sdk#174]
Adding the 

```
<dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-smallrye-opentracing</artifactId>
</dependency>
```

to the joke sample project produces this log at startup :

```
================================================================================
CDI: programmatic lookup problem detected
-----------------------------------------
At least one bean matched the required type and qualifiers but was marked as unused and removed during build

Stack frame: io.quarkus.resteasy.common.runtime.QuarkusConstructorInjector.construct(QuarkusConstructorInjector.java:35)
Required type: class io.quarkus.smallrye.opentracing.runtime.QuarkusSmallRyeTracingDynamicFeature
Required qualifiers: [@javax.enterprise.inject.Default()]
Removed beans:
        - CLASS bean  [types=[interface javax.ws.rs.container.DynamicFeature, class io.quarkus.smallrye.opentracing.runtime.QuarkusSmallRyeTracingDynamicFeature], qualifiers=null]
Solutions:
        - Application developers can eliminate false positives via the @Unremovable annotation
        - Extensions can eliminate false positives via build items, e.g. using the UnremovableBeanBuildItem
        - See also https://quarkus.io/guides/cdi-reference#remove_unused_beans
        - Enable the DEBUG log level to see the full stack trace
================================================================================
```

and NPE :

```
2021-12-18 13:19:32,807 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.lang.NullPointerException
        at io.quarkus.smallrye.opentracing.runtime.QuarkusSmallRyeTracingDynamicFeature.configure(QuarkusSmallRyeTracingDynamicFeature.java:26)
        at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.<init>(ClientInvoker.java:87)
        at org.jboss.resteasy.client.jaxrs.internal.proxy.ProxyBuilderImpl.createClientInvoker(ProxyBuilderImpl.java:50)
        at org.jboss.resteasy.client.jaxrs.internal.proxy.ProxyBuilderImpl.build(ProxyBuilderImpl.java:149)
        at org.jboss.resteasy.client.jaxrs.internal.proxy.ProxyBuilderImpl.build(ProxyBuilderImpl.java:108)
        at io.quarkus.restclient.runtime.QuarkusRestClientBuilder.build(QuarkusRestClientBuilder.java:339)
        at io.quarkus.restclient.runtime.RestClientBase.create(RestClientBase.java:72)
        at io.quarkiverse.operatorsdk.samples.joke.JokeService_797272e0f58967d9bc86ed502c8088f2ee0fa963_Synthetic_Bean.create(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.JokeService_797272e0f58967d9bc86ed502c8088f2ee0fa963_Synthetic_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at io.quarkiverse.operatorsdk.samples.joke.JokeService_797272e0f58967d9bc86ed502c8088f2ee0fa963_Synthetic_Bean.get(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.JokeService_797272e0f58967d9bc86ed502c8088f2ee0fa963_Synthetic_Bean.get(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.JokeController_Bean.create(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.JokeController_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:18)
        at io.quarkiverse.operatorsdk.samples.joke.JokeController_ClientProxy.arc$delegate(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.JokeController_ClientProxy.arc_contextualInstance(Unknown Source)
        at io.quarkus.arc.runtime.ClientProxyUnwrapper.apply(ClientProxyUnwrapper.java:11)
        at io.quarkiverse.operatorsdk.runtime.QuarkusConfigurationService.unwrap(QuarkusConfigurationService.java:74)
        at io.quarkiverse.operatorsdk.runtime.QuarkusConfigurationService.getConfigurationFor(QuarkusConfigurationService.java:63)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer.operator(OperatorProducer.java:36)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer_Subclass.operator$$superforward1(Unknown Source)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer_Subclass$$function$$1.apply(Unknown Source)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.proceed(InvocationInterceptor.java:62)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor.monitor(InvocationInterceptor.java:51)
        at io.quarkus.arc.runtime.devconsole.InvocationInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer_Subclass.operator(Unknown Source)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer_ProducerMethod_operator_b4530a54321662e5391d4f625ba48900cce797ee_Bean.create(Unknown Source)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer_ProducerMethod_operator_b4530a54321662e5391d4f625ba48900cce797ee_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer_ProducerMethod_operator_b4530a54321662e5391d4f625ba48900cce797ee_Bean.get(Unknown Source)
        at io.quarkiverse.operatorsdk.runtime.OperatorProducer_ProducerMethod_operator_b4530a54321662e5391d4f625ba48900cce797ee_Bean.get(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.Main_Bean.create(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.Main_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:18)
        at io.quarkiverse.operatorsdk.samples.joke.Main_ClientProxy.arc$delegate(Unknown Source)
        at io.quarkiverse.operatorsdk.samples.joke.Main_ClientProxy.run(Unknown Source)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:125)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:67)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
        at io.quarkiverse.operatorsdk.samples.joke.Main.main(Main.java:31)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

I guess that the way controller are registered to CDI or resolved from make the rest client invisible to quarkus bean trimmer.

adding `quarkus.arc.unremovable-types=io.quarkus.smallrye.opentracing.runtime.QuarkusSmallRyeTracingDynamicFeature` to application.properties as a workaround.

It seems this happens since Quarkus 2.0.0 as it does not occured in my quarkus:1.13.7/operator:1.19 apps. Someone on Quarkus repo raised a similar issue https://github.com/quarkusio/quarkus/issues/18105 but got closed because the PO was actually not using a rest client.

**Comment by geoand** [comment]
The root cause is probably the same as reported in https://github.com/quarkusio/quarkus/issues/22605

**Comment by metacosm** [comment]
Should be resolved in 2.0.3.

**Comment by sclorng** [comment]
Test OK. Thanks a lot.
